geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java

```diff
@@ -758,9 +758,7 @@ public void prepareNewMember(InternalDistributedMember sender, PersistentMemberI
 
       // The oldId and newId could be the same if the member is retrying a GII. See bug #42051
       if (oldId != null && !oldId.equals(newId)) {
-        if (initialized) {
-          memberRemoved(oldId, false);
-        }
+        memberRemoved(oldId, false);
```

Comment:
I am not sure implication of not to wait for initialization; but looking at the code above this line; where a newId is added without checking for initialization; this seems to be fine.